src/io/flutter/actions/FlutterSdkAction.java

```diff
@@ -30,19 +32,36 @@
   public void actionPerformed(AnActionEvent event) {
     final Project project = DumbAwareAction.getEventProject(event);
 
-    final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
-    if (sdk == null) {
-      showMissingSdkDialog(project);
-      return;
-    }
+    if(enableActionInBazelContext() && project != null) {
+      // See if the Bazel workspace provides a script.
+      final Workspace workspace = WorkspaceCache.getInstance(project).getNow();
+      if (workspace != null) {
+          FlutterInitializer.sendAnalyticsAction(this);
+          FileDocumentManager.getInstance().saveAllDocuments();
+          startCommandInBazelContext(project, workspace);
+      }
+    } else {
+      final FlutterSdk sdk = project != null ? FlutterSdk.getFlutterSdk(project) : null;
+      if (sdk == null) {
+        showMissingSdkDialog(project);
+        return;
+      }
 
-    FlutterInitializer.sendAnalyticsAction(this);
-    FileDocumentManager.getInstance().saveAllDocuments();
-    startCommand(project, sdk, PubRoot.forEventWithRefresh(event));
+      FlutterInitializer.sendAnalyticsAction(this);
+      FileDocumentManager.getInstance().saveAllDocuments();
+      startCommand(project, sdk, PubRoot.forEventWithRefresh(event));
+    }
   }
 
   public abstract void startCommand(@NotNull Project project, @NotNull FlutterSdk sdk, @Nullable PubRoot root);
 
+  public void startCommandInBazelContext(@NotNull Project project, @NotNull Workspace workspace) {
+  }
```

Comment:
Not all actions can/should be run in the bazel context.  The idea here is to have methods override this method and enableActionInBazelContext if they should have the opportunity to run in this mode.

Added this commment to startCommandInBazelContext:


  /**
   * Implemented by actions which are used in the Bazel context ({@link #enableActionInBazelContext()} returns true), by default this method
   * is a no-op.
   */